### PR TITLE
subclass 5xx errors

### PR DESCRIPTION
**Pull request by SimonSchick**
Allows for less specific code in fe2.

### CODE REVIEW

**Comment by UnwrittenFun** [comment]
502 + 503 I would think.

Thanks btw 😄

**Comment by SimonSchick** [comment]
Oh right, my bad, will adjust.

File: src/adapters/errors/ServerError.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { IModel, IModelCtor, IPersistentAdapter } from '../../Types';
+import { IResponse } from '../http';
+import { AdapterResponseError } from './AdapterResponseError';
+
+/**
+ * Occurs when credentials were correct but 2fa failed.
```
Comment (UnwrittenFun) [inline src/adapters/errors/ServerError.ts]: outdated description

File: src/adapters/errors/UnavailableError.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { IModel, IModelCtor, IPersistentAdapter } from '../../Types';
+import { IResponse } from '../http';
+import { ServerError } from './ServerError';
+
+/**
+ * Occurs when credentials were correct but 2fa failed.
```
Comment (UnwrittenFun) [inline src/adapters/errors/UnavailableError.ts]: and here

**Pull request merged**